components/proxy/src/main/java/com/hotels/styx/proxy/interceptors/TcpTunnelRequestRejector.java

```diff
@@ -0,0 +1,46 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.proxy.interceptors;
+
+import com.hotels.styx.api.Eventual;
+import com.hotels.styx.api.HttpInterceptor;
+import com.hotels.styx.api.HttpMethod;
+import com.hotels.styx.api.HttpResponseStatus;
+import com.hotels.styx.api.LiveHttpRequest;
+import com.hotels.styx.api.LiveHttpResponse;
+
+import static com.hotels.styx.api.LiveHttpResponse.response;
+
+/**
+ * A built-in interceptor to reject  connections that try to use the CONNECT method to establish a TCP tunnel
+ * (see issue #312 https://github.com/HotelsDotCom/styx/issues/312).
+ */
+public class TcpTunnelRequestRejector implements HttpInterceptor {
+
+
+
+    @Override
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
+
+        if (HttpMethod.CONNECT.equals(request.method())) {
+            return Eventual.of(response(HttpResponseStatus.METHOD_NOT_ALLOWED).build());
+        } else {
+            return chain.proceed(request);
+        }
+    }
+
+
```

Comment:
Overall this is what needs to be done. However please follow styx coding standards, specifically use static imports with METHOD_NOT_ALLOWED, and CONNECT. Also I would remove the extra empty lines.